### PR TITLE
[AutoScheduler] Fix deserization of workload registry entry

### PR DESCRIPTION
**Pull request by vinx13**
Added the same condition used in [`serialize`](https://github.com/vinx13/tvm/blob/d90edc06cc12f558ad92489b58aecc0bd20b12be/python/tvm/auto_scheduler/workload_registry.py#L226) 

cc @comaniac @jcf94 @dlexplorer

### CODE REVIEW

**Review by shingjan — COMMENTED** [review]
LGTM. Will defer the approval to the codeowner(s). This should be able to unblock #8492 by fixing the xgboost failed tests.

**Review by tqchen — APPROVED** [review]

**Review by comaniac — APPROVED** [review]
Oops. Thanks for the fix!

**Review by junrushao1994 — COMMENTED** [review]
Thanks @vinx13 for nailing down the issue!

**Review by junrushao1994 — APPROVED** [review]

**Review by jcf94 — APPROVED** [review]

**Pull request merged**
